### PR TITLE
feat: support dynamic app db backend flag

### PR DESCRIPTION
**Pull request by yihuang**
Build rocksdb enabled binary:
```
ENABLE_ROCKSDB=true COSMOS_BUILD_OPTIONS=rocksdb make build
```

Enable `app-db-backend` in `app.config`, or use  cli flag:`ethermintd start --app-db-backend "rocksdb"`

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ### Improvements
 
 * (feemarket) [\#1165](https://github.com/evmos/ethermint/pull/1165) Add hint in specs about different gas terminology for gas in Cosmos and Ethereum.
+* (cli) []() Add custom app db backend flag.
```
Comment (fedekunze) [inline CHANGELOG.md:53]: update

File: server/start.go [hunk 2]
```diff
@@ -45,6 +45,8 @@ import (
 	srvflags "github.com/evmos/ethermint/server/flags"
 )
 
+const FlagAppDBBackend = "app-db-backend"
```
Comment (fedekunze) [inline server/start.go:48]: Let's move this to the flags directory

Comment (fedekunze) [reply server/start.go:48]: and add a comment

File: cmd/ethermintd/root.go [hunk 3]
```diff
@@ -210,6 +210,11 @@ func (a appCreator) newApp(logger tmlog.Logger, db dbm.DB, traceStore io.Writer,
 	}
 
 	snapshotDir := filepath.Join(cast.ToString(appOpts.Get(flags.FlagHome)), "data", "snapshots")
+	err = os.MkdirAll(snapshotDir, os.ModePerm)
+	if err != nil {
```
Comment (fedekunze) [inline cmd/ethermintd/root.go:214]: nit
```suggestion
	if err := os.MkdirAll(snapshotDir, os.ModePerm); err != nil {
```

**Review by fedekunze — COMMENTED** [review]
ACK. Minor comments

File: cmd/ethermintd/root.go (hunk 3 above)
Comment (yihuang) [reply cmd/ethermintd/root.go]: done

File: server/start.go (hunk 2 above)
Comment (yihuang) [reply server/start.go:48]: done

File: CHANGELOG.md (hunk 1 above)
Comment (yihuang) [reply CHANGELOG.md]: done

File: server/start.go [hunk 4]
```diff
@@ -45,6 +45,8 @@ import (
...
+
```
Comment (fedekunze) [inline server/start.go:49]:
```suggestion
```

File: server/start.go [hunk 5]
```diff
@@ -141,6 +143,7 @@ which accepts a path for the resulting pprof file.
 	cmd.Flags().Uint64(server.FlagPruningInterval, 0, "Height interval at which pruned heights are removed from disk (ignored if pruning is not 'custom')")
 	cmd.Flags().Uint(server.FlagInvCheckPeriod, 0, "Assert registered invariants every N blocks")
 	cmd.Flags().Uint64(server.FlagMinRetainBlocks, 0, "Minimum block height offset during ABCI commit to prune Tendermint blocks")
+	cmd.Flags().String(srvflags.AppDBBackend, "", "The type of database for application and snapshots databases")
```
Comment (fedekunze) [inline server/start.go:146]: what's the default here?

**Review by fedekunze — APPROVED** [review]

File: server/start.go (hunk 5 above)
Comment (yihuang) [reply server/start.go:144]: there's a fallback chain:
- the old compile flag `github.com/cosmos/cosmos-sdk/types.DBBackend`
- the `db_backend` in `config.toml`

**Pull request merged**